### PR TITLE
Adds SQL to travis. Other travis optimizations

### PR DESCRIPTION
**Pull request by Cyberboss**
Discovered this while working on BSQL. Better tested with than without

In addition, rust-g only builds once and each job only pulls the dependencies it needs instead of them all pulling everything.

### CODE REVIEW

**Comment by chesse20** [comment]
add this please ive wanted this SO MUCH :yum:

**Comment by Cyberboss** [comment]
Done

**Comment by Cyberboss** [comment]
For real

**Comment by SpaceManiac** [comment]
Is this correct?
* `BUILD_TOOLS=true` builds tgui
* `BUILD_TESTING=true` builds the game with all maps
* `BUILD_TESTING=false` builds the game with unit tests and runs it

**Comment by Cyberboss** [comment]
Yes, you might want to add "Also builds rustg" to that last one

File: .travis.yml [hunk 1]
```diff
@@ -9,35 +9,51 @@ env:
...
+            - libc6-i386
+            - libstdc++6:i386
+      cache:
+        directories:
+          - $HOME/BYOND-${BYOND_MAJOR}.${BYOND_MINOR}
+    - env:
+      - BUILD_TESTING=false
+      - BUILD_TOOLS=false
+      addons:
+        mariadb: '10.2'
```
Comment (Jordie0608) [inline .travis.yml]: Mariadb 10.3 is the latest stable release, probably better to use that if travis supports it.

**Comment by Jordie0608** [comment]
This only tests the normal schema, is it reasonable to also run the prefixed schema through as well or will that get messy?

File: .travis.yml (hunk 1 above)
Comment (Cyberboss) [reply .travis.yml]: Readme specifies 10.2 as minimum supported so I don't think we should change that

**Pull request merged**